src/CommonITILObject.php

```diff
@@ -906,12 +906,13 @@ public function canAdminActors()
      */
     public function canAssign()
     {
-        if (
-            isset($this->fields['is_deleted']) && ($this->fields['is_deleted'] == 1)
-            || isset($this->fields['status']) && in_array($this->fields['status'], $this->getClosedStatusArray())
-        ) {
+        $is_closed = isset($this->fields['status']) && in_array($this->fields['status'], static::getClosedStatusArray());
+        $is_deleted = isset($this->fields['is_deleted']) && (int) $this->fields['is_deleted'] === 1;
```

Comment:
For this particular line, a isDeleted() exists in CommonDBTM.
To generalize, I think we may have state methods for CommonITILObject like `isSolved`, `isClosed`, etc